### PR TITLE
RHOAIENG-1119: Don't rely on imagestream in image info task

### PR DESCRIPTION
**Pull request by grdryn**
This PR stacks upon #215 as another preparatory commit for [RHOAIENG-1119](https://issues.redhat.com//browse/RHOAIENG-1119). It will be held until that one is merged, but it can still be reviewed, by just reviewing the new commit(s).

## Description
With this change, we now get info for the image from the image metadata using `skopeo inspect`, rather than from the ImageStream resource in OpenShift.

## How Has This Been Tested?

- `make go-test` runs successfully
- Example pipelineruns executed manually with success

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: pipelines/tekton/aiedge-e2e/tasks/retrieve-build-image-info.task.yaml
Comment (MarianMacik) [inline pipelines/tekton/aiedge-e2e/tasks/retrieve-build-image-info.task.yaml:43]: Just FYI, when I tried this, when inspecting `docker://` image, it returns uncompressed size, when using `docker-daemon:sha256:<sha256-of-local-docker-registry>` it seems to return the full size.
E.g.:
`skopeo inspect docker://docker.io/library/fedora@sha256:830a6bcb5c9b30bfa77479b747a904bed02151e9de7ceeb8a1d117b574571fa4` (fedora:latest)
returns 
``` json
"Size": 64614010
```
whereas
`skopeo inspect docker-daemon:sha256:5f878605acd6f24f3f11e4085544caf61b046378b31d6b3aa0677a7a3dbe0a17`
which is my local IMAGE ID for the same image returns
```json
"Size": 182162432
```
and docker inspect for that same image returns
```json
"Size": 176658692
```
The last 2 outputs are from local image, the first one is for a remote, compressed image.

I think for now it's OK. Just in case we need to have constraints about image size we should be more careful what we are measuring.

Comment (grdryn) [reply pipelines/tekton/aiedge-e2e/tasks/retrieve-build-image-info.task.yaml:43]: Yeah, that's interesting. I was actually initially just going to omit this bit of information (and maybe one or two others also), since I'm not sure why we want it or how we'll use it.

It does appear to be the compressed size that we were getting previously: I imported an image into an imagestream & compared. E.g.:

```
# import image to imagestream
oc import-image fedora --from=quay.io/fedora/fedora --all --confirm

# then check the size of a tag in the imagestream
oc get -o jsonpath='{.image.dockerImageMetadata.Size}' imagestreamtag/fedora:39
68672016

# compare that to the new approach of adding layer sizes in original repo
echo $(($(skopeo inspect --format '{{range .LayersData}}+{{.Size}}{{end}}' docker://quay.io/fedora/fedora:39)))
68670700
```

^there's a slight discrepancy (~1KiB), which I'd guess is the manifest referring to the layers?

Comment (MarianMacik) [reply pipelines/tekton/aiedge-e2e/tasks/retrieve-build-image-info.task.yaml:43]: This metric is for the future when we are planning to check the real image size and have a test for it, if the image is above the limit, the pipeline would likely fail or something similar.

Also for my other images with more than one layer the numbers are slightly off, roughly by 5 MB. Interestingly skopeo reports bigger sum than docker inspect. But it may be also due to rounding or different way of calculation by skopeo.

File: pipelines/tekton/aiedge-e2e/aiedge-e2e.pipeline.yaml
Comment (StevenTobin) [inline pipelines/tekton/aiedge-e2e/aiedge-e2e.pipeline.yaml:245]: Should this be configurable? Would we expect this to change.

Comment (grdryn) [reply pipelines/tekton/aiedge-e2e/aiedge-e2e.pipeline.yaml:245]: Yes, it will become configurable as part of #218. This PR/commit is just preparation for that

**Comment by grdryn** [comment]
Going to close and re-open in an attempt to get GitHub to automatically update the PR view (it's out of date and showing a commit that's already on main).

**Comment by grdryn** [comment]
> Going to close and re-open in an attempt to get GitHub to automatically update the PR view (it's out of date and showing a commit that's already on main).

Wow, that actually worked. This commit should now be even easier to review

**Comment by Sara4994** [comment]
This /lgtm when test goes green

**Comment by grdryn** [comment]
@Sara4994 thanks for the review! You don't need to wait for the test status check to go green before adding the /lgtm (or approve using the review feature), since it won't automatically merge until it has both the lgtm label and the test job passes. So please just /lgtm :slightly_smiling_face:

**Review by biswassri — COMMENTED** [review]
/lgtm

**Review by Sara4994 — APPROVED** [review]

**Pull request merged**
